Sum of digits of a number.py

```diff
@@ -1,7 +1,31 @@
-q=0
-n=int(input("Enter Number: "))
-while(n>0):
- r=n%10
- q=q+r
- n=n//10
-print("Sum of digits is: "+str(q))
+import sys
+
+
+def get_integer():
+    for i in range(3,0,-1):
+        num = input("enter a number:")
+        if num.isnumeric():
+            num = int(num)
+            return num
+        else:
+            print("enter integer only")
+            print(f'{i-1} chances are left')
+            continue
+
+
+def addition(num):
+    Sum=0
+    if type(num) is type(None):
+        print("Try again!")
+        sys.exit()
+    while num > 0:
+        digit = int(num % 10)
+        Sum = Sum + digit
+        num = num / 10
+    return Sum
+       
+
+number = get_integer()
+Sum = addition(number)
+print(f'sum of digits of {number} is {Sum}')
```

Comment:
also use,
if __name__ == "__main__":
    function1()
    function2()
    function3()
    
    
    
    
    
    
    
 # Something adding to the bottom of the file is good, practice.

Reply:
> also use, if **name** == "**main**": function1() function2() function3()
> 
> # Something adding to the bottom of the file is good, practice.

Could you explain this a little bit more @NitkarshChourasia.

Reply:
Try chatGPT or google.com, but do implement it.

Reply:
@sarayusreeyadavpadala https://www.tutorialspoint.com/What-does-the-if-name-main-do-in-Python

Reply:
> Try chatGPT or google.com, but do implement it.

Okay @NitkarshChourasia